### PR TITLE
meta: fix the allocator batch size compute logic

### PR DESCRIPTION
**Pull request by AilinKid**
### What problem does this PR solve?

Problem Summary: fix logic in allocator batch size  computation

### What is changed and how it works?

What's Changed:
when the local cache size if not enough for allocN, do the new batchSize computation based on the global new base in the txn.

so we postpone the NextStep adjustment to meta txn and store it after that.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code

### Release note 
- meta: fix the allocator batch size compute logic

### CODE REVIEW

File: meta/autoid/autoid.go [hunk 1]
```diff
@@ -640,21 +645,19 @@ func (alloc *allocator) alloc4Signed(tableID int64, n uint64, increment, offset
...
 			var err1 error
 			newBase, err1 = getAutoIDByAllocType(m, alloc.dbID, tableID, alloc.allocType)
 			if err1 != nil {
 				return err1
 			}
+			// CalcNeededBatchSize calculates the total batch size needed on global base.
+			n1 = CalcNeededBatchSize(newBase, int64(n), increment, offset, alloc.isUnsigned)
+			// Although the step is customized by user, we still need to make sure nextStep is big enough for insert batch.
+			if nextStep < n1 {
+				nextStep = n1
```
Comment (djshow832) [inline meta/autoid/autoid.go:659]: It was `n1*2` before but it's `n1` now. Is that right?

Comment (AilinKid) [reply meta/autoid/autoid.go:659]: yes, allocate what you have asked for is correct

**Pull request merged**